### PR TITLE
businesstime!

### PR DESCRIPTION
**Pull request by jjidt**
- moves user and project fetching/creation to businesstime layer
- moves user password functions to coordinator layer since this is touching the user model and also accessing the password generator/validator
- adds ono library for Error creation, which allows us to easily set error.code property anywhere in code, and then send that status to the api consumer.  Also allows us to re-throw errors with updated messages/data without losing stack trace
- All data coming out of BusinessTime :tm: are now POJOs

### CODE REVIEW

File: api/src/controllers/projects.js [hunk 1]
```diff
@@ -1,61 +1,56 @@
...
   router.get('/:id', getProject)
 }
 
 const getProjects = async function(req, res) {
-  const db = getDb()
-  let projects
   try {
-    projects = await db.model('Project').findAll({}, { raw: true })
+    const projects = await BusinessProject.findAll()
+    throw new Error('blarp')
```
Comment (dheniges) [inline api/src/controllers/projects.js]: v. informative

File: api/src/coordinators/user.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import passwords from '../libs/passwords'
+import userBusiness from '../businesstime/user'
```
Comment (dheniges) [inline api/src/coordinators/user.js]: I think we should stick with either businessUser or userBusiness. In the projects controller you have businessProject

**Review by dheniges — CHANGES_REQUESTED** [review]
It all makes sense to me. I think you left an extraneous error in there, plus one naming nitpick

File: api/src/controllers/projects.js (hunk 1 above)
Comment (jjidt) [reply api/src/controllers/projects.js]: haha

Comment (dheniges) [inline api/src/controllers/projects.js]: blarp is still here?

Comment (jjidt) [reply api/src/controllers/projects.js]: haha blarp forever

**Review by dheniges — DISMISSED** [review]
Blarp is dead, long live Blarp! 👍

**Comment by jjidt** [comment]
@dheniges realized I forgot status codes and error condition tests, added those if you wanna take a quick glance at that latest commit

**Review by dheniges — APPROVED** [review]
Updated tests still pass 👍

**Pull request merged**